### PR TITLE
Fix parse error when using named import "as" with flow parser

### LINKED ISSUE #13520: [Bug]:  Parse error when using named import "as" with `babel-plugin-transform-flow-strip-types`
### 💻

- [ ] Would you like to work on a fix?

### How are you using Babel?

Other (Next.js, Gatsby, vue-cli, ...)

### Input code

A file containing a named import using `as` or `As` results in a parse error when using the `babel-plugin-transform-flow-strip-types`. This plugin is used by `babelJest` when transpiling some tests including ES6 modules.

For example, this excerpt belongs to a file defining a named import with name `as`, which is absolutely valid.

```js
import { historyIcon as as } from "../shapes/history.js";
```

_How to get this file:_ 
```
wget `npm v @cds/core dist.tarball` && tar -xOzf core-5.4.1.tgz package/icon/collections/essential.js 
```

### Configuration file name

_No response_

### Configuration

Using the default config in CRA (create-react-application):
https://github.com/facebook/create-react-app/blob/master/packages/react-scripts/config/jest/babelTransform.js#L26-L37

### Current and expected behavior

Current behavior when running `yarn test` in my app:

```console
SyntaxError: /home/agamez/Desktop/kubeapps/dashboard/node_modules/@cds/core/icon/collections/essential.js: Unexpected token, expected "," (1:3239)

    > 58 | import{historyIcon as as}from"../shapes/history.js";
         |                    ^

      at Object._raise (node_modules/@babel/parser/src/parser/error.js:97:45)
      at Object.raiseWithData (node_modules/@babel/parser/src/parser/error.js:92:17)
      at Object.raise (node_modules/@babel/parser/src/parser/error.js:41:17)
      at Object.unexpected (node_modules/@babel/parser/src/parser/util.js:161:16)
      at Object.expect (node_modules/@babel/parser/src/parser/util.js:138:28)
      at Object.parseNamedImportSpecifiers (node_modules/@babel/parser/src/parser/statement.js:2364:14)
      at Object.parseImport (node_modules/@babel/parser/src/parser/statement.js:2145:39)
      at Object.parseStatementContent (node_modules/@babel/parser/src/parser/statement.js:265:25)
      at Object.parseStatement (node_modules/@babel/parser/src/parser/statement.js:176:17)
      at Object.parseStatement (node_modules/@babel/parser/src/plugins/flow/index.js:1815:26)
```

Expected behavior: no parse errors whatsoever.

### Environment

```
  System:
    OS: Linux 5.8 Ubuntu 20.04.2 LTS (Focal Fossa)
  Binaries:
    Node: 14.15.0 - ~/.nvm/versions/node/v14.15.0/bin/node
    Yarn: 1.22.5 - /usr/bin/yarn
    npm: 7.15.1 - ~/.nvm/versions/node/v14.15.0/bin/npm
```

### Possible solution

When enabling (using the default true value) `flow` in CRA [it loads the `@babel/plugin-transform-flow-strip-types'` plugin](https://github.com/facebook/create-react-app/blob/master/packages/babel-preset-react-app/create.js#L111) which is  [loading getting the AST given a file](https://github.com/babel/babel/blob/main/packages/babel-plugin-transform-flow-strip-types/src/index.js#L28). Here is where the parse error is returned.

My guess is that the parser used by this plugin is performing a wrong tokenization in the particular case of a named import with the same value as the predecessor token `as`.

### Additional context

I wasn't able to reproduce the parse error using the `babelParser.parse` method manually, so I guess it has something to do with the `babel-plugin-transform-flow-strip-types` plugin.

More context: https://github.com/kubeapps/kubeapps/pull/3042#issuecomment-870053634

### PR DESCRIPTION
**Pull request by doing-art**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | Fixes #13520  
| Patch: Bug Fix?          | 👍
| Major: Breaking Change?  |
| Minor: New Feature?      |
| Tests Added + Pass?      | 👍
| Documentation PR Link    | 
| Any Dependency Changes?  |
| License                  | MIT

Update logic of the parseImportSpecifier function in the flow parser plugin to fix the issue #13520. Add related tests. 

<a href="https://gitpod.io/#https://github.com/babel/babel/pull/13659"></a>

### CODE REVIEW

File: packages/babel-parser/src/plugins/flow/index.js [hunk 1]
```diff
@@ -2686,10 +2686,14 @@ export default (superClass: Class<Parser>): Class<Parser> =>
...
         /*:: invariant(firstIdent instanceof N.Node) */
-        isBinding = true;
         specifier.imported = firstIdent;
         specifier.importKind = null;
-        specifier.local = cloneIdentifier(specifier.imported);
+        if (this.eatContextual("as")) {
+          specifier.local = this.parseIdentifier();
+        } else {
+          isBinding = true;
+          specifier.local = specifier.imported.__clone();
```
Comment (nicolo-ribaudo) [inline packages/babel-parser/src/plugins/flow/index.js:2695]: We are moving away from `__clone()` (https://github.com/babel/babel/commit/d3a7cd5e8d5f99fb0b465a5c1d7209d5e1e6ab03). Can you use this instead?
```suggestion
          specifier.local = cloneIdentifier(specifier.imported);
```

Also, do you think we can merge these lines with lines 2673-2678? Maybe by doing something like this:
```js
if (this.isContextual("as") && !this.isLookaheadContextual("as")) { // Line 2649
  // ...
} else {
  if (specifierTypeKind !== null && (this.match(tt.name) || this.state.type.keyword)) {
    // `import {type foo`
    specifier.imported = this.parseIdentifier(true);
    specifier.importKind = specifierTypeKind;
  } else {
    specifier.imported = firstIdent;
    specifier.importKind = null;
  }
  
  if (this.eatContextual("as")) {
    // ...
  } else {
    // ...
  }
}
```

**Review by fedeci — COMMENTED** [review]
Can you add a test for 
```js
import { type foo as bar } from 'name'
```

File: packages/babel-parser/src/plugins/flow/index.js (hunk 1 above)
Comment (doing-art) [reply packages/babel-parser/src/plugins/flow/index.js]: Merged the lines as you suggested. Thank you.

**Review by fedeci — APPROVED** [review]
Thanks!

**Review by nicolo-ribaudo — APPROVED** [review]
Awesome!

**Pull request merged**